packages/story-editor/src/components/library/panes/media/media3p/providerPanel.js

```diff
@@ -60,10 +63,60 @@ const ProviderWrapper = styled.div`
 `;
 
 function ProviderPanel({ providerType, isActive, searchTerm, ...rest }) {
+  const { updateElementsByResourceId } = useStory(({ actions }) => ({
+    updateElementsByResourceId: actions.updateElementsByResourceId,
+  }));
+
   const { insertElement } = useLibrary((state) => ({
     insertElement: state.actions.insertElement,
   }));
 
+  const {
+    actions: { registerUsage },
+  } = useMedia3pApi();
+
+  /**
+   * If the resource has a register usage url then the fact that it's been
+   * inserted needs to be registered as per API provider policies.
+   *
+   * @param {Object} resource The resource to attempt to register usage.
+   */
+  const handleRegisterUsage = useCallback(
+    (resource) => {
+      if (resource?.attribution?.registerUsageUrl) {
+        registerUsage({
+          registerUsageUrl: resource.attribution.registerUsageUrl,
+        });
+      }
+    },
+    [registerUsage]
+  );
+
+  // Update the base color for an existing media3p resource on the page.
+  // Like useDetectBaseColor, but without updating the local media library.
+  const updateBaseColor = useCallback(
+    async (resource) => {
+      const { id, type, src, poster } = resource;
+      const imageSrc = type === 'image' ? src : poster;
+      if (!imageSrc) {
+        return;
+      }
+      try {
+        const color = await getMediaBaseColor(imageSrc);
```

Comment:
You need to ensure that url is passed through the proxy.

Reply:
This is 3p media.

3p media never needs the proxy.

Reply:
I would still personally put it through the proxy anyway, just incase.

Reply:
Strongly disagree; the proxy is only needed for the use case of hotlinked media.